Tests/SKCoreTests/BuildSystemManagerTests.swift

```diff
@@ -264,7 +263,13 @@ final class BuildSystemManagerTests: XCTestCase {
     await bsm.registerForChangeNotifications(for: h, language: .c)
     assertEqual(await bsm.buildSettingsInferredFromMainFile(for: h, language: .c), bs.map[cpp1]!)
 
-    mainFiles.mainFiles[h] = Set([cpp2])
+    await mainFiles.setMainFiles(
+      mainFiles: [
+        h: Set([cpp2]),
+        cpp1: Set([cpp1]),
+        cpp2: Set([cpp2]),
+      ]
+    )
```

Comment:
We shouldn’t need to re-state the main files for `cpp1` and `cpp2` here. What you think of having the following function instead of `setMainFiles`?

```swift
func updateMainFiles(for file: DocumentURI, to mainFiles: Set<DocumentURI>)
```

Reply:
true, let me update the code